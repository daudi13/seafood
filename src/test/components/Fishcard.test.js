/* eslint linebreak-style: ["error", "windows"] */
import { render, screen } from '@testing-library/react';
import FishCard from '../../components/Fishcard/Fishcard';
import 'react-intersection-observer';

describe('test Fishcard component', () => {
  test('check whether the component renders correctly', () => {
    render(<FishCard name="test name1" img="test image" weight="test kg" onClick={() => { }} />);
    expect(screen.getByText('test name1'));
    expect(screen.getByText('test kg'));
  });
});
