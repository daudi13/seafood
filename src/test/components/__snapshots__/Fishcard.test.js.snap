// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Fishcard component check whether the component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="fishcard"
        style="background-image: url(fish.img);"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium pointer css-i4bv87-MuiSvgIcon-root"
          data-testid="ArrowCircleRightOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
          />
        </svg>
        <div
          class="fishname"
        >
          <h3>
            fish.name
          </h3>
          <small>
            fish.maxWeightkgs
          </small>
        </div>
        <div
          class="overlay"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="fishcard"
      style="background-image: url(fish.img);"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium pointer css-i4bv87-MuiSvgIcon-root"
        data-testid="ArrowCircleRightOutlinedIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M22 12c0-5.52-4.48-10-10-10S2 6.48 2 12s4.48 10 10 10 10-4.48 10-10zM4 12c0-4.42 3.58-8 8-8s8 3.58 8 8-3.58 8-8 8-8-3.58-8-8zm12 0-4 4-1.41-1.41L12.17 13H8v-2h4.17l-1.59-1.59L12 8l4 4z"
        />
      </svg>
      <div
        class="fishname"
      >
        <h3>
          fish.name
        </h3>
        <small>
          fish.maxWeightkgs
        </small>
      </div>
      <div
        class="overlay"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
